tensorflow/compiler/xla/service/gpu/horizontal_loop_fusion.cc

```diff
@@ -62,28 +62,45 @@ class HorizontalLoopFusionImpl {
   StatusOr<bool> Run();
 
  private:
-  Status Fuse(absl::Span<HloInstruction*> fused_fusion_instrs);
+  Status Fuse(absl::Span<HloInstruction*> fused_fusion_instrs, bool sliced_input_fusion,
+              std::vector<HloInstruction*>& to_fuse_candidates);
 
-  // Horizontally fuses `fused_fusion_instrs`. It is required that each of
-  // `fused_fusion_instrs` is a kLoop fusion. Also, we require their numbers of
+  // Horizontally fuses `fused_fusion_instrs` into kInput computation. It is required that 
+  // each of `fused_fusion_instrs` is a kLoop fusion. Also, we require their numbers of
   // outputs to be the same, so that each output will be fused/concatenated with
   // the same number of outputs from other fused fusion instrs. Then, all the
   // fused outputs still have the same shapes for kernel generation.
   //
   // Returns the fused computation in `uniq_computation` and the operands that
   // are used by `uniq_computation`.
-  Status CreateFusedComputation(
+  Status CreateSliceInputFusedComputation(
       absl::Span<HloInstruction*> fused_fusion_instrs,
       std::unique_ptr<HloComputation>* uniq_computation,
       std::vector<HloInstruction*>* bound_operands);
 
+  // Horizontally fuses `fused_fusion_instrs` into kLoop computation.  
+  // Returns the fused computation in `uniq_computation` and the operands that
+  // are used by `uniq_computation`.
+  Status CreateLoopFusedComputation(
+      absl::Span<HloInstruction*> fused_fusion_instrs,
+      std::unique_ptr<HloComputation>* uniq_computation,
+      std::vector<HloInstruction*>* bound_operands);
+
+  // Horizontal fuse the operands of consumer instruction, `sliced_input_fusion` 
+  // controls whether kInput or kLoop type fused instruction want to be created. 
+  // `to_fuse_candidates` is the instruction stack that we want to 
+  // try horizontally fuse its operands, when we create a new fusion instruction,
+  // we push it to the stack in hope to further fuse its operands.
+  StatusOr<bool> FuseConsumerOperands(HloInstruction* consumer, bool sliced_input_fusion,
+                                      std::vector<HloInstruction*>& to_fuse_candidates);
+
   // FusionCandidates collects profitable candidates for a given consumer
   // instruction. GetNextSpanOfFusions() can then be iteratively invoked to
   // acquire the next set of fusion candidates based on some heuristics.
   class FusionCandidates {
    public:
-    explicit FusionCandidates(HloInstruction* consumer)
-        : fusible_instrs_(), pos_(0) {
+    explicit FusionCandidates(HloInstruction* consumer, bool sliced_input_fusion)
+        : fusible_instrs_(), sliced_input_fusion_(sliced_input_fusion), pos_(0) {
```

Comment:
Put the "sliced_input_fusion_" initialization at the end of the list to avoid the "reorder-ctor" compiler warning.

Reply:
Fixed